src/configurationProvider.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+import TelemetryReporter from "vscode-extension-telemetry";
+import * as commands from "./commands";
+
+export class JavaConfigurationProvider implements vscode.DebugConfigurationProvider {
+    private _resolving: boolean = false;
+
+    constructor(private _reporter: TelemetryReporter) {
+    }
+
+    // Returns an initial debug configurations based on contextual information.
+    public provideDebugConfigurations(folder: vscode.WorkspaceFolder | undefined, token?: vscode.CancellationToken):
+        vscode.ProviderResult<vscode.DebugConfiguration[]> {
+        return [{
+            type: "java",
+            name: "Debug (Launch)",
+            request: "launch",
+            mainClass: "",
+            args: "",
+        }, {
+            type: "java",
+            name: "Debug (Attach)",
+            request: "attach",
+            hostName: "localhost",
+            port: 0,
+        }];
+    }
+
+    // Try to add all missing attributes to the debug configuration being launched.
+    public resolveDebugConfiguration(folder: vscode.WorkspaceFolder | undefined, config: vscode.DebugConfiguration, token?: vscode.CancellationToken):

```

Comment:
This method is not marked async yet call async meethods

Reply:
This is VSCode API built-in methods. No need to change it to async. because the result `vscode.ProviderResult<vscode.DebugConfiguration>` could be T or Thenable<T>.

Reply:
That's the point. If you mark as async, then the caller will know how to deal with it. If you don't provide the same behavior as method signature, the behavior will be undefined in some cases.